score-data-format/src/main/java/io/cloudslang/content/json/services/JsonService.java

```diff
@@ -0,0 +1,154 @@
+        char quote = '\"';   //  the default quote character used to specify json member names and string value according to the json specification
+        for (char c : jsonString.toCharArray()) {
+            if (c == '\'' || c == '\"') {
+                quote = c;
+                break;
+            }
+        }
+        return quote;
+    }
+
+    private void removeEmptyElementsFromMap(Map<String, Object> json) {
+        LinkedHashMap<String, Object> copyJson = new LinkedHashMap<String, Object>(json);
+        for (Object element : copyJson.values()) {
+            if (element == null) {
+                json.values().remove(element);
+            } else if (element instanceof String) {
+                if (element.equals("")) {
+                    json.values().remove(element);
+                }
+            } else if (element instanceof JSONArray) {
+                if (((JSONArray) element).isEmpty()) {
+                    json.values().remove(element);
+                } else {
+                    removeEmptyElementFromJsonArray((JSONArray) element);
+                }
+            } else if (element instanceof LinkedHashMap) {
+                if (((LinkedHashMap) element).isEmpty()) {
+                    json.values().remove(element);
+                } else {
+                    removeEmptyElementsFromMap((Map<String,Object>) element);
+                }
+            }
+        }
+    }
+
+    private void removeEmptyElementFromJsonArray(JSONArray jsonArray) {
+        JSONArray copyJsonArray = (JSONArray) jsonArray.clone();
+        for (Object element : copyJsonArray) {
+            if (element == null) {
+                jsonArray.remove(element);
+            } else if (element instanceof String) {
+                if (element.equals("")) {
+                    jsonArray.remove(element);
+                }
+            } else if (element instanceof JSONArray) {
+                if (((JSONArray) element).isEmpty()) {
+                    jsonArray.remove(element);
+                } else {
+                    removeEmptyElementFromJsonArray((JSONArray) element);
+                }
+            } else if (element instanceof LinkedHashMap) {
+                if (((LinkedHashMap) element).isEmpty()) {
+                    jsonArray.remove(element);
+                } else {
+                    removeEmptyElementsFromMap((LinkedHashMap) element);
+                }
+            }
+        }
+    }
+
+    private String replaceUnescapedOccurrencesOfCharacterInText(String text, char toReplace, char newChar) {
+        char[] charArrayText = text.toCharArray();
+        for (int i = 0; i < charArrayText.length; i++) {
+            if(shouldCharacterBeReplace(charArrayText,toReplace,i))  {
+                charArrayText[i] = newChar;
+            }
+        }
+
+        return String.valueOf(charArrayText);
+    }
+
+    private boolean shouldCharacterBeReplace(char[] characters,char characterToReplace,int characterPosition) {
```

Comment:
maybe rename method to "shouldCharacterBeReplaced" or "shouldTheCharacterBeReplaced"